src/rgw/rgw_keystone.cc

```diff
@@ -203,10 +203,15 @@ int KeystoneToken::parse(CephContext * const cct,
 
 bool RGWKeystoneTokenCache::find(const string& token_id, KeystoneToken& token)
 {
-  lock.Lock();
+  Mutex::Locker l(lock);
+  return find_locked(token_id, token);
+}
+
+bool RGWKeystoneTokenCache::find_locked(const string& token_id, KeystoneToken& token)
+{
+  assert(lock.is_locked_by_me());
```

Comment:
ceph ships with asserts enabled, so this could be costly;  if you believe the code is correct, can we skip asserting?